jwst/saturation/saturation.py

```diff
@@ -160,6 +165,24 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
                                         irs2_mask, detector)
             # check for saturation
             flag_temp = np.where(sci_temp >= sat_thresh, SATURATED, 0)
+            if group == 2:
+                # Identify groups which we wouldn't expect to saturate by the third group,
+                # on the basis of the first group
+                mask = sci_temp[ints, 0, ...] / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                # Identify groups with suspiciously large values in the second group
+                mask &= sci_temp[ints, 1, ...] > sat_thresh / len(read_pattern[1])
```

Comment:
Likewise here, this will just return element [0,1] of the 2d sci_temp array.

Reply:
I think we also need something to grow this mask by n_pix_grow_sat using the adjacency_sat routine.  For pixels bordering pixels that saturate in group 3, the saturation flag is set too.  Those pixels might not be intrinsically bright enough to trigger the 'suspiciously large' criterion, but will likewise be affected by the CR and restricted to only fitting a ramp to groups 1+2.

Reply:
@drlaw1558 thank you for the comments! I'm committing the changes soon. Can I please ask you to take look again when they are in on this branch as well?